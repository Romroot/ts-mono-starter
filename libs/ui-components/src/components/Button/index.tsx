import React from 'react'
import { StyledButton } from './Button.styles'

export interface ButtonProps {
  /**
   * This is custom description
   */
  children?: React.ReactNode
  onClick?(event: any): void
  text?: React.ReactNode
}

/**
 * Button component
 */
export const Button: React.FC<ButtonProps> =  ({ children, ...props }) => {
  return (
    <StyledButton type="button" {...props}>
      {children}
    </StyledButton>
  )
}

Button.defaultProps = {
  text: 'Default text',
}

export default Button
