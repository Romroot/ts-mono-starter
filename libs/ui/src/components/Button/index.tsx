import { ReactNode, FC } from "react";
import styles from "./button.module.css";
export interface ButtonProps {
  children?: ReactNode;
  onClick?(): void;
  text?: ReactNode;
}

export const Button: FC<ButtonProps> = ({ children, ...props }) => {
  return (
    <button type="button" {...props} className={styles.button}>
      {children}
    </button>
  );
};

export default Button;
