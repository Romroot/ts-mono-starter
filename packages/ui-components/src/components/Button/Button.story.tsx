import React from 'react'
import { action } from '@storybook/addon-actions'

export default {
  title: 'Button',
}

// eslint-disable-next-line
const Button = ({ children, ...props }: any): any => <button {...props}>{children}</button>

export const text = () => <Button onClick={action('clicked')}>Hello Button</Button>

export const emoji = () => (
  <Button onClick={action('clicked')}>
    <span role="img" aria-label="so cool">
      😀 😎 👍 💯
    </span>
  </Button>
)
