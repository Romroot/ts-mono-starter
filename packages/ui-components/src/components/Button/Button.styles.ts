import styled from 'styled-components'

export const StyledButton = styled.button`
  border-radius: 2px;
  background: darkred;
  color: white;
`
