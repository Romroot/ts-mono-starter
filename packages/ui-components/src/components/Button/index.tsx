import React from 'react'

const b: any = {}

export default function Button() {
  const aa = b?.d?.g?.g?.()
  return (
    <button type="button" disabled>
      HELLO WORLD BUTTON {aa}
    </button>
  )
}
