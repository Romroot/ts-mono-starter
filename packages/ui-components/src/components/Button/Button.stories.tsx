import React from 'react'
import Button from './index'

export default {
  title: 'Button Title',
  info: { inline: true, header: false },
  component: Button,
  parameters: {
    componentSubtitle: 'To trigger an operation.',
  },
}

export const text = () => <Button>Hello Button</Button>
