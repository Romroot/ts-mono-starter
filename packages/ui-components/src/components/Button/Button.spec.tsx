import React from 'react'
import renderer from 'react-test-renderer'
import Button from './'

describe('Button', () => {
  it('match snapshot', () => {
    const tree = renderer.create(<Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
