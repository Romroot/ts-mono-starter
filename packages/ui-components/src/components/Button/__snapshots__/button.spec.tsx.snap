// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `
<button
  disabled={true}
  type="button"
>
  HELLO WORLD BUTTON 
</button>
`;
